### PR TITLE
Chore: Remove CreateUserForTests()

### PR DESCRIPTION
**Pull request by papagian**
**What is this feature?**

It removes `CreateUserForTests()` that was used only by the tests.
It makes tests more deterministic by reducing assumption that the main org exists; reduces usage of hard coded OrgIDs (set to 1)

**Why do we need this feature?**

We suspect that `CreateUserForTests()` can be responsible from the flaky tests.
It's better that the tests are aligned with the actual implementation.

**Who is this feature for?**

[Add information on what kind of user the feature is for.]

**Which issue(s) does this PR fix?**:

Fixes #

**Special notes for your reviewer**:

this should only affect tests not actual implementation

### CODE REVIEW

File: pkg/services/org/orgimpl/store_test.go [hunk 1]
```diff
@@ -542,20 +570,21 @@ func TestIntegration_SQLStore_GetOrgUsers(t *testing.T) {
...
+				OrgID: o.ID,
 				User: &user.SignedInUser{
-					OrgID:       1,
+					OrgID:       o.ID,
 					Permissions: map[int64]map[string][]string{1: {accesscontrol.ActionOrgUsersRead: {accesscontrol.ScopeUsersAll}}},
 				},
 			},
-			expectedNumUsers: 10,
+			// out of the 10 created users all but the 1st are added to the organization
+			expectedNumUsers: 9,
```
Comment (papagian) [inline pkg/services/org/orgimpl/store_test.go:580]: not sure what's exactly the purpose of this block but it was already there so I have kept it:
https://github.com/grafana/grafana/blob/323a4f9499ebfd65640933ca9a8d690081612400/pkg/services/org/orgimpl/store_test.go#L637-L644

Comment (sakjur) [reply pkg/services/org/orgimpl/store_test.go:580]: I _think_ the first user is added as an admin somewhere else maybe?

File: pkg/tests/api/azuremonitor/azuremonitor_test.go [hunk 2]
```diff
@@ -105,7 +105,7 @@ func TestIntegrationAzureMonitor(t *testing.T) {
 		// nolint:gosec
 		resp, err := http.Post(u, "application/json", buf1)
 		require.NoError(t, err)
-		require.Equal(t, http.StatusBadRequest, resp.StatusCode)
+		//require.Equal(t, http.StatusBadRequest, resp.StatusCode)
```
Comment (idafurjes) [inline pkg/tests/api/azuremonitor/azuremonitor_test.go:108]: is this on purpose or leftover?

**Review by idafurjes — APPROVED** [review]
LGTM

File: pkg/tests/api/azuremonitor/azuremonitor_test.go (hunk 2 above)
Comment (ying-jeanne) [inline pkg/tests/api/azuremonitor/azuremonitor_test.go:108]:
```suggestion
```

Comment (papagian) [reply pkg/tests/api/azuremonitor/azuremonitor_test.go:108]: by mistake! good catch

File: pkg/services/org/orgimpl/store_test.go (hunk 1 above)
Comment (papagian) [reply pkg/services/org/orgimpl/store_test.go:580]: hmm 🤔  the deleted `CreateUserForTests` used to do that:
https://github.com/grafana/grafana/blob/3468804ff5f3eb542df414680e1838c4183d05ea/pkg/services/user/userimpl/user.go#L426-L446
thank you, I will fix the test appropriately to restore the old behaviour

**Review by sakjur — APPROVED** [review]

**Review by Jguer — APPROVED** [review]

**Review by mildwonkey — APPROVED** [review]
ALL THE APPROVALS

File: pkg/cmd/grafana-cli/commands/conflict_user_command_test.go [hunk 3]
```diff
@@ -107,17 +108,20 @@ func TestBuildConflictBlock(t *testing.T) {
...
 			sqlStore := db.InitTestDB(t)
-			usrSvc := setupTestUserService(t, sqlStore)
 			if sqlStore.GetDialect().DriverName() != ignoredDatabase {
 				for _, u := range tc.users {
-					cmd := user.CreateUserCommand{
-						Email: u.Email,
-						Name:  u.Name,
-						Login: u.Login,
-						OrgID: int64(testOrgID),
+					userStore := userimpl.ProvideStore(sqlStore, sqlStore.Cfg)
```
Comment (Jguer) [inline pkg/cmd/grafana-cli/commands/conflict_user_command_test.go:113]: looks a bit weird to create a user store for each user?

**Pull request merged**